#175636848 User should be able to request one way trip

bigelio opened the pull request:
#### What does this PR do?
User should be able to request one way trip
#### Description of Task to be completed?
- Defining CRUD for one way trip request operations.
- Defining validation schema for validating inputs included in trip request.
- Validating user inputs on new trip request and on edit a submitted request.
- Defining a model for one way trip request.
- Updating api documentations with new routes.
- Writing tests for all new apis that are not tested.
#### How should this be manually tested?
run `git clone https://github.com/atlp-rwanda/tech-strikers-bn-backend.git`
run `cd  tech-strikers-bn-backend`
run `git checkout ft-one-way-trip-request-175636848`
run `npm install`
run `npm run migrations`
run `npm run dev`

You can use any tool to access the following endpoints but I recommend to use **postman**

Get all requests `/api/v1/trip/one-way`
Get a request by id `/api/v1/trip/one-way/:id`
Create a request `/api/v1/trip/one-way`
Edit a request `/api/v1/trip/one-way`
Delete a request `/api/v1/trip/one-way`

#### Any background context you want to provide?
No
#### What are the relevant pivotal tracker stories?
[#175636848](https://www.pivotaltracker.com/story/show/175636848)
#### Screenshots (if appropriate)
- Requesting a trip
<img width="799" alt="Capture" src="https://user-images.githubusercontent.com/48241489/104526579-70153880-560b-11eb-9e7b-47b86844c6fe.PNG">

- Getting all request
<img width="797" alt="Capture1" src="https://user-images.githubusercontent.com/48241489/104526614-84593580-560b-11eb-8115-58092b52efd4.PNG">

- Editing a trip request
<img width="804" alt="Capture2" src="https://user-images.githubusercontent.com/48241489/104526635-989d3280-560b-11eb-80e2-2e7a74a718ae.PNG">

fabriceumuhire commented on src/database/models/oneWayTrip.js:
```diff
@@ -0,0 +1,34 @@
+// const {
+//   Model
+// } = require('sequelize');
+module.exports = (sequelize, DataTypes) => {
```
Can you change this to `export default` instead of `module.exports` ?

hervebutera commented on src/database/models/index.js:
```diff
@@ -4,7 +4,13 @@ import Sequelize from "sequelize";
 
 const basename = _basename(__filename);
-const env = process.env.NODE_ENV || "development";
+let env;
```
line 7 was better  just have a `NODE_ENV`  variable in `.env` file and give it value of the environment you are in.  Because those conditions you've defined don't consider the `production` environment.

bigelio replied:
sure!

hervebutera commented on test/welcome/user.test.js:
```diff
@@ -0,0 +1,21 @@
+        done();
+      });
+  });
+});
```
A file name related to the file content is better!

hervebutera commented on src/database/migrations/20210107065115-create-location.js:
```diff
@@ -0,0 +1,31 @@
+      },
+      city: {
+        type: Sequelize.STRING,
+        defaultValue: true,
```
default value `true` on `city` and `country` fields doesn't sound like a better option!

hervebutera reviewed (commented):
It's better naming a migration `xxxxxx-create-trip.js` instead of `xxxxx-create-onewayTrip.js`  because it is a common file for other tasks. And all other files that are common across tasks!

hervebutera committed:
Merge pull request #39 from atlp-rwanda/ft-user-role-setting-175636852

#175636852 Add User Role-Setting for Super Administrator to Assign Roles to Users Using Their Emails

hervebutera committed:
feat(trip request): User shoul be able to request one way trip

- Defining CRUD for one way trip request operations
- Defining validation schema for validating inputs included in trip request
- Defining a model for one way trip request
- Added service functions to query database
- Setting up testing environment and with all routes set for testing

[finished #175636848]

bigelio committed:
feat(one-way-trip): One way trip request

Doing rebase on main feature for tracking news changes

bigelio committed:
feat(trip request): one way trip request

Rebasing for changes of resetting password
